lib/GLPI/Agent/Task/Inventory/Generic/Remote_Mgmt/MeshCentral.pm

```diff
@@ -0,0 +1,96 @@
+package GLPI::Agent::Task::Inventory::Generic::Remote_Mgmt::MeshCentral;
+
+use strict;
+use warnings;
+
+use parent 'GLPI::Agent::Task::Inventory::Module';
+
+use English qw(-no_match_vars);
+
+use GLPI::Agent::Tools;
+
+sub isEnabled {
+    my (%params) = @_;
+    my $logger    = $params{logger};
+
+    if (OSNAME eq 'MSWin32') {
+
+        GLPI::Agent::Tools::Win32->use();
+
+        my $key = getRegistryKey(
+            path        => "HKEY_LOCAL_MACHINE/SOFTWARE/Open Source/Mesh Agent",
+            required    => [ qw/NodeId/ ],
+            maxdepth    => 1,
+            logger      => $params{logger}
+        );
+
+        return $key && (keys %$key);
+
+    } elsif (OSNAME eq 'darwin') {
+        return canRun('defaults') && Glob(
+            "/Library/LaunchDaemons/meshagent*.plist"
+        );
+    }
+
+    return Glob('/etc/systemd/system/meshagent.service');
+}
+
+sub doInventory {
+    my (%params) = @_;
+
+    my $inventory = $params{inventory};
+    my $logger    = $params{logger};
+
+    my $nodeId = _getNodeId()
```

Comment:
If you keep `$logger` and need it in other functions for debugging, you need to make a change here and check other comments:
```suggestion
    my $nodeId = _getNodeId(logger => $logger)
```